Code generation appears to be broken

Since the latest update to `urfave/cli`, it looks like we need to make some adjustments to the sources:

![image](https://github.com/user-attachments/assets/00007f51-4ee6-4726-8a27-7d688cab5a88)

```
# github.com/urfave/cli-docs/v3
/var/tmp/go/pkg/mod/github.com/urfave/cli-docs/v3@v3.0.0-alpha5/docs.go:273:16: flag.GetValue undefined (type cli.DocGenerationFlag has no field or method GetValue)
/var/tmp/go/pkg/mod/github.com/urfave/cli-docs/v3@v3.0.0-alpha5/docs.go:400:21: flag.GetValue undefined (type cli.DocGenerationFlag has no field or method GetValue)
```

```
github.com/urfave/cli-docs/v3 v3.0.0-alpha5 h1:H1oWnR2/GN0dNm2PVylws+GxSOD6YOwW/jI5l78YfPk=
github.com/urfave/cli-docs/v3 v3.0.0-alpha5/go.mod h1:AIqom6Q60U4tiqHp41i7+/AB2XHgi1WvQ7jOFlccmZ4=
github.com/urfave/cli/v3 v3.0.0-alpha9.2 h1:CL8llQj3dGRLVQQzHxS+ZYRLanOuhyK1fXgLKD+qV+Y=
github.com/urfave/cli/v3 v3.0.0-alpha9.2/go.mod h1:FnIeEMYu+ko8zP1F9Ypr3xkZMIDqW3DR92yUtY39q1Y=
```

I'll take care of it ASAP

--- a/.github/workflows/main.yml
+++ b/.github/workflows/main.yml
@@ -10,13 +10,13 @@ jobs:
     strategy:
       matrix:
         os: [ubuntu-latest, macos-latest, windows-latest]
-        go: [1.18.x, 1.19.x, 1.20.x]
+        go: [stable, oldstable]
     runs-on: ${{ matrix.os }}
     steps:
-      - uses: actions/setup-go@v4
+      - uses: actions/setup-go@v5
         with:
           go-version: ${{ matrix.go }}
-      - uses: actions/checkout@v3
+      - uses: actions/checkout@v4
         with:
           fetch-depth: 0
       - run: make

--- a/docs.go
+++ b/docs.go
@@ -6,6 +6,7 @@ import (
 	"fmt"
 	"io"
 	"os"
+	"reflect"
 	"regexp"
 	"runtime"
 	"sort"
@@ -277,7 +278,7 @@ func prepareFlags(
 // flagDetails returns a string containing the flags metadata
 func flagDetails(flag cli.DocGenerationFlag) string {
 	description := flag.GetUsage()
-	value := flag.GetValue()
+	value := getFlagDefaultValue(flag)
 	if value != "" {
 		description += " (default: " + value + ")"
 	}
@@ -404,7 +405,7 @@ func (tt tabularTemplate) PrepareFlags(flags []cli.Flag) []cliTabularFlagTemplat
 			Usage:      tt.PrepareMultilineString(flag.GetUsage()),
 			EnvVars:    flag.GetEnvVars(),
 			TakesValue: flag.TakesValue(),
-			Default:    flag.GetValue(),
+			Default:    getFlagDefaultValue(flag),
 		}
 
 		if boolFlag, isBool := appFlag.(*cli.BoolFlag); isBool {
@@ -553,4 +554,35 @@ func (tabularTemplate) Prettify(s string) string {
 	s = strings.Trim(s, " \n")                                   // trim spaces and newlines
 
 	return s + "\n" // add an extra newline
+}
+
+// getFlagDefaultValue returns the default value of a flag. Previously, the [cli.DocGenerationFlag] interface included
+// a GetValue string method, but it was removed in https://github.com/urfave/cli/pull/1988.
+// This function serves as a workaround, attempting to retrieve the value using the removed method; if that fails, it
+// tries to obtain it via reflection (the [cli.FlagBase] still has a Value field).
+func getFlagDefaultValue(f cli.DocGenerationFlag) string {
+	if !f.TakesValue() {
+		return ""
+	}
+
+	if v, ok := f.(interface{ GetValue() string }); ok {
+		return v.GetValue()
+	}
+
+	var ref = reflect.ValueOf(f)
+	if ref.Kind() != reflect.Ptr {
+		return ""
+	} else {
+		ref = ref.Elem()
+	}
+
+	if ref.Kind() != reflect.Struct {
+		return ""
+	}
+
+	if val := ref.FieldByName("Value"); val.IsValid() && val.Type().Kind() != reflect.Bool {
+		return fmt.Sprintf("%v", val.Interface())
+	}
+
+	return ""
 }

--- a/go.mod
+++ b/go.mod
@@ -4,14 +4,13 @@ go 1.18
 
 require (
 	github.com/cpuguy83/go-md2man/v2 v2.0.2
-	github.com/stretchr/testify v1.8.4
-	github.com/urfave/cli/v3 v3.0.0-alpha4
+	github.com/stretchr/testify v1.9.0
+	github.com/urfave/cli/v3 v3.0.0-alpha9.2
 )
 
 require (
 	github.com/davecgh/go-spew v1.1.1 // indirect
 	github.com/pmezard/go-difflib v1.0.0 // indirect
 	github.com/russross/blackfriday/v2 v2.1.0 // indirect
-	github.com/xrash/smetrics v0.0.0-20201216005158-039620a65673 // indirect
 	gopkg.in/yaml.v3 v3.0.1 // indirect
 )

